lib/wpc_proto/internal_modules/proto_otap.c

```diff
@@ -30,27 +38,113 @@ void Proto_otap_close(void)
 
 }
 
+static app_res_e handle_scratchpad_chunk(uint8_t * chunk,
+                                         size_t chunk_size,
+                                         uint32_t offset,
+                                         size_t total_size,
+                                         uint32_t seq)
+{
+    app_res_e res;
+    uint8_t status;
+    LOGD("Uploading scratchpad chunk: size %u, offset %u, total_size %u, seq %u\n",
+                            chunk_size,
+                            offset,
+                            total_size,
+                            seq);
+
+    /* Check if it is first chunk and no other load ongoing */
+    if (offset == 0)
+    {
+        if (m_scratchpad_load_current_seq != INVALID_CURRENT_SEQ)
+        {
+            LOGW("Previous upload not finshed: old seq %u, new seq %u\n",
+                                m_scratchpad_load_current_seq,
+                                seq);
+
+            /* Only a log, keep going with it */
+        }
+
+        if (WPC_get_stack_status(&status) == APP_RES_OK && status == 0)
+        {
+            /* Stack must be stoped */
+            /* No check required as next step will anyway fail */
+            WPC_stop_stack();
+            m_restart_after_load = true;
+        }
+
+        m_scratchpad_load_current_seq = seq;
+        res = WPC_start_local_scratchpad_update(total_size, seq);
+        if (res != APP_RES_OK)
+        {
+            return res;
```

Comment:
if the start failed, could be good to leave (or set) m_scratchpad_load_current_seq to INVALID_CURRENT_SEQ